### PR TITLE
Social share for promos

### PR DESCRIPTION
**Pull request by paulsancer**
New layout for promos:

-  Top header shows what company's is the promo
-  Top header right shows **share button** for Facebook, Twitter and WhatsApp (mobile only). It just shares the URL of the promo.
-  Image is now in the middle of the paper taking full width of it, Facebook like.
-  Description after the image.
-  Action buttons at the bottom: Details and Map.

Please take a look, not sure if I should've change that :zipper_mouth_face: .

### CODE REVIEW

File: client/src/components/Home/PromotionSingleResult.jsx [hunk 1]
```diff
@@ -1,25 +1,72 @@
...
-        <div className="col-xs-12 col-sm-9">
-          <h3>{title}</h3>
-          <p>{description}</p>
+        <h4 className="title">
+          {title}
+          <Chip style={styles.chip} labelStyle={styles.chipLabel}>
+            Comida
+          </Chip>
+        </h4>
+        <small className="availability">Válido hasta el 12/05/2018</small>
```
Comment (BernabeFelix) [inline client/src/components/Home/PromotionSingleResult.jsx]: Comment and add a ToDo to implement if we have time

File: client/src/components/Home/PromotionSingleResult.jsx [hunk 2]
```diff
@@ -1,25 +1,72 @@
...
-      <div className="row">
-        <div className="col-xs-12 col-sm-3">
-          <div className="img-container">
-            <img className="img-responsive" src={imgUrl} alt="" />
+  <Paper className="promotion-single-result" zDepth={2}>
+    <div className="row">
+      <div className="col-xs-12 col-sm-12">
+        <h4 className="company">
+          <Link to="/pizza-hut">Pizza Hut &copy;</Link>
+          <ShareMenu title={title} url={getFullUrl(url({ id }))} />
```
Comment (BernabeFelix) [inline client/src/components/Home/PromotionSingleResult.jsx]: To check in call:
- Why url as prop and not with router HOC
- Why url is not in proptypes

File: client/src/components/Home/PromotionSingleResult.jsx [hunk 3]
```diff
@@ -1,25 +1,72 @@
...
 const PromotionSingleResult = ({ description, id, imgUrl, url, title }) => (
-  <Link to={url({ id })}>
-    <Paper className="promotion-single-result" zDepth={2}>
-      <div className="row">
-        <div className="col-xs-12 col-sm-3">
-          <div className="img-container">
-            <img className="img-responsive" src={imgUrl} alt="" />
+  <Paper className="promotion-single-result" zDepth={2}>
+    <div className="row">
+      <div className="col-xs-12 col-sm-12">
```
Comment (BernabeFelix) [inline client/src/components/Home/PromotionSingleResult.jsx]: cols size inherit size from last col definition, in this case all cols size above `col-xs-12` will be 12 , so, `col-sm-12' is redundant

File: client/src/components/Home/PromotionSingleResult.jsx [hunk 4]
```diff
@@ -1,25 +1,72 @@
...
+        <h4 className="company">
+          <Link to="/pizza-hut">Pizza Hut &copy;</Link>
```
Comment (BernabeFelix) [inline client/src/components/Home/PromotionSingleResult.jsx]: Why is this not a url?

File: client/src/components/Home/PromotionSingleResult.jsx [hunk 5]
```diff
@@ -1,25 +1,72 @@
...
+            Comida
+          </Chip>
+        </h4>
+        <small className="availability">Válido hasta el 12/05/2018</small>
+        <p className="description">{description}</p>
+        <div className="actions">
+          <RisedButton
+            label="Detalles"
+            primary
+            containerElement={<Link to={url({ id })} />}
```
Comment (BernabeFelix) [inline client/src/components/Home/PromotionSingleResult.jsx]: Invert Son and Parent, 
```
<Link />
 <RaisedButton ...
```

File: client/src/components/Home/PromotionSingleResult.jsx [hunk 6]
```diff
@@ -1,25 +1,72 @@
...
+            style={styles.buttonRoot}
+          />
+          <FlatButton
```
Comment (BernabeFelix) [inline client/src/components/Home/PromotionSingleResult.jsx]: Remove map button

File: client/src/components/Home/ShareMenu.jsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+} from 'react-share';
+
+const styles = {
+  shareIconStyles: {
+    fontSize: '24px',
+    color: 'rgb(117, 117, 117)'
+  },
+  iconMenu: {
+    iconStyle: {
+      color: '#757575',
```
Comment (BernabeFelix) [inline client/src/components/Home/ShareMenu.jsx]: Replace by `$gray`

File: client/src/components/Home/ShareMenu.jsx [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+import {
+  FacebookShareButton,
+  TwitterShareButton,
+  WhatsappShareButton
+} from 'react-share';
+
+const styles = {
+  shareIconStyles: {
+    fontSize: '24px',
+    color: 'rgb(117, 117, 117)'
```
Comment (BernabeFelix) [inline client/src/components/Home/ShareMenu.jsx]: Check if can be replaced by a $color

**Comment by BernabeFelix** [comment]
Rename src/api/auth/auth.jsx by auth.js

File: client/src/utils/url.js [hunk 9]
```diff
@@ -1,3 +1,5 @@
-export const hasSlashAtTheEnd = str => {
-  return str.lastIndexOf('/') === str.length - 1;
-};
+import settings from '../../settings.json';
+
+export const hasSlashAtTheEnd = str => str.lastIndexOf('/') === str.length - 1;
+
+export const getRootUrl = () => settings.rootUrl;
```
Comment (BernabeFelix) [inline client/src/utils/url.js]: Replace by window.location.href

**Review by BernabeFelix — CHANGES_REQUESTED** [review]

File: client/src/components/Home/ShareMenu.jsx (hunk 7 above)
Comment (paulsancer) [reply client/src/components/Home/ShareMenu.jsx]: done

File: client/src/components/Home/ShareMenu.jsx (hunk 8 above)
Comment (paulsancer) [reply client/src/components/Home/ShareMenu.jsx]: done

File: client/src/utils/url.js (hunk 9 above)
Comment (paulsancer) [reply client/src/utils/url.js]: done

File: client/src/components/Home/PromotionSingleResult.jsx (hunk 5 above)
Comment (paulsancer) [reply client/src/components/Home/PromotionSingleResult.jsx]: done

**Review by BernabeFelix — APPROVED** [review]

**Pull request merged**